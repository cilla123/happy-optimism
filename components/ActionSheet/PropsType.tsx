export type Action = {
  text: string,
  theme?: 'default' | 'primary' | 'success' | 'warning' | 'error',
  className?: string,
  onClick?: () => void,
};

export default interface PropsType {
  visible?: boolean;
  shape?: 'radius';
  spacing?: boolean;
  actions: Action[];
  cancelText?: string;
  onMaskClick?: () => void;
  onCancel?: () => void;
}
