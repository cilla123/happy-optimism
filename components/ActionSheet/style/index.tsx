import '../../style/';
import '../../Popup/style';
import './index.scss';
