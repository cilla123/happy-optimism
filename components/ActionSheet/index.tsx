import React, { PureComponent } from 'react';
import classnames from 'classnames';
import PropsType from './PropsType';
import Popup from '../Popup';

export interface ActionSheetProps extends PropsType {
  prefixCls?: string;
  className?: string;
}

export default class ActionSheet extends PureComponent<ActionSheetProps, {}> {

  static defaultProps = {
    prefixCls: 'za-actionsheet',
    visible: false,
    spacing: false,
    actions: [],
    cancelText: '取消',
  };

  renderActions = (action, index) => {
    const { prefixCls } = this.props;
    const actionCls = classnames(`${prefixCls}-item`, {
      [`theme-${action.theme}`]: !!action.theme,
    });
    return <a key={+index} className={actionCls} onClick={action.onClick}>{action.text}</a>;
  }

  renderCancel = () => {
    const { prefixCls, onCancel, cancelText } = this.props;
    return (typeof onCancel === 'function') && (
      <div className={`${prefixCls}-cancel`}>
        <a className={`${prefixCls}-item`} onClick={onCancel}>{cancelText}</a>
      </div>
    );
  }

  render() {
    const { prefixCls, className, shape, spacing, visible, onMaskClick, actions } = this.props;
    const cls = classnames(`${prefixCls}`, className, {
      [`shape-${shape}`]: !!shape,
      [`${prefixCls}-spacing`]: spacing,
    });

    return (
      <Popup visible={visible} onMaskClick={onMaskClick}>
        <div className={cls}>
          <div className={`${prefixCls}-actions`}>
            {actions.map(this.renderActions)}
          </div>
          {this.renderCancel()}
        </div>
      </Popup>
    );
  }
}
