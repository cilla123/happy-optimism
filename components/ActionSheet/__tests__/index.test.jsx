import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ActionSheet from '../index';

describe('ActionSheet', () => {
  const props = {
    visible: true,
    actions: [
      {
        text: '操作一',
        onClick: jest.fn(),
      },
      {
        text: '操作二',
        onClick: jest.fn(),
      },
      {
        theme: 'error',
        text: '操作三',
        onClick: jest.fn(),
      },
    ],
    onMaskClick: jest.fn(),
  };

  it('renders correctly', () => {
    const wrapper = mount(<ActionSheet {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // it('click cancel button', () => {
  //   props.onCancel = jest.fn();
  //   const wrapper = mount(<ActionSheet {...props} />);
  //   document.body.find('.za-actionsheet-cancel').childAt(0).simulate('click');
  //   expect(props.onCancel).toBeCalled();
  // });
});
