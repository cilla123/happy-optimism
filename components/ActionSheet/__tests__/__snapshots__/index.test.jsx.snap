// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionSheet renders correctly 1`] = `
<ActionSheet
  actions={
    Array [
      Object {
        "onClick": [Function],
        "text": "操作一",
      },
      Object {
        "onClick": [Function],
        "text": "操作二",
      },
      Object {
        "onClick": [Function],
        "text": "操作三",
        "theme": "error",
      },
    ]
  }
  cancelText="取消"
  onMaskClick={[Function]}
  prefixCls="za-actionsheet"
  spacing={false}
  visible={true}
>
  <Popup
    animationDuration={200}
    autoClose={false}
    direction="bottom"
    mask={true}
    maskType="normal"
    onMaskClick={[Function]}
    prefixCls="za-popup"
    stayTime={3000}
    visible={true}
  >
    <div>
      <Portal
        getContainer={[Function]}
      >
        <div
          className="za-popup"
        >
          <div
            className="za-popup-wrapper za-popup-wrapper-bottom"
            style={
              Object {
                "WebkitTransitionDuration": "200ms",
                "transitionDuration": "200ms",
              }
            }
          >
            <div
              className="za-actionsheet"
            >
              <div
                className="za-actionsheet-actions"
              >
                <a
                  className="za-actionsheet-item"
                  key="0"
                  onClick={[Function]}
                >
                  操作一
                </a>
                <a
                  className="za-actionsheet-item"
                  key="1"
                  onClick={[Function]}
                >
                  操作二
                </a>
                <a
                  className="za-actionsheet-item theme-error"
                  key="2"
                  onClick={[Function]}
                >
                  操作三
                </a>
              </div>
            </div>
          </div>
          <Mask
            className=""
            onClose={[Function]}
            prefixCls="za-mask"
            style={
              Object {
                "WebkitAnimationDuration": "200ms",
                "animationDuration": "200ms",
              }
            }
            type="normal"
            visible={true}
          >
            <div
              className="za-mask normal"
              onClick={[Function]}
              style={
                Object {
                  "WebkitAnimationDuration": "200ms",
                  "animationDuration": "200ms",
                }
              }
            />
          </Mask>
        </div>
      </Portal>
    </div>
  </Popup>
</ActionSheet>
`;
